homeassistant/components/cover/deconz.py

```diff
@@ -61,11 +67,14 @@ def __init__(self, cover):
     async def async_added_to_hass(self):
         """Subscribe to covers events."""
         self._cover.register_async_callback(self.async_update_callback)
-        self.hass.data[DATA_DECONZ].deconz_ids[self.entity_id] = \
-            self._cover.deconz_id
+        self.gateway.deconz_ids[self.entity_id] = self._cover.deconz_id
+        self.unsub_dispatcher = async_dispatcher_connect(
+            self.hass, DECONZ_REACHABLE, self.async_update_callback)
 
     async def async_will_remove_from_hass(self) -> None:
         """Disconnect cover object when removed."""
+        if self.unsub_dispatcher is not None:
+            self.unsub_dispatcher()
         self._cover.remove_callback(self.async_update_callback)
```

Comment:
You called the register_async_callback, maybe call other unregister_async_callback? Consistent naming looks nicer and improves readability

Reply:
I'll think about it, not related to this PR though ;)